src/overlays/actors/ovl_Bg_Ikana_Mirror/z_bg_ikana_mirror.c

```diff
@@ -28,99 +30,365 @@ const ActorInit Bg_Ikana_Mirror_InitVars = {
+    this->dyna.actor.flags |= ACTOR_FLAG_20;
+    this->actionFunc = BgIkanaMirror_EmitLight;
+}
+
+void BgIkanaMirror_EmitLight(BgIkanaMirror* this, PlayState* play) {
+    s32 i;
+
+    for (i = 0; i < ARRAY_COUNT(this->colliderQuad); i++) {
+        if ((this->colliderQuad[i].base.atFlags & AT_HIT)) {
+            this->colliderQuad[i].base.atFlags &= ~AT_HIT;
+        }
+    }
+
+    if (this->lightEmissionAlpha < 155) {
+        this->lightEmissionAlpha += 100;
+    } else {
+        this->lightEmissionAlpha = 255;
+    }
+
+    if (this->lightAbsorptionAlpha >= 61) {
+        this->lightAbsorptionAlpha -= 60;
+    } else {
+        this->lightAbsorptionAlpha = 0;
+    }
 
-extern ColliderTrisElementInit D_80B7FF50[9];
-extern ColliderTrisInit D_80B8016C;
-extern ColliderQuadInit D_80B8017C;
-extern ColliderQuadInit D_80B801FC;
-extern InitChainEntry D_80B8021C[];
+    if (this->timer > 0) {
+        this->timer--;
 
-extern UNK_TYPE D_06001E18;
-extern UNK_TYPE D_06002358;
+        for (i = 0; i < ARRAY_COUNT(this->colliderQuad); i++) {
+            CollisionCheck_SetAT(play, &play->colChkCtx, &this->colliderQuad[i].base);
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Mirror/func_80B7F730.s")
+    } else {
+        this->dyna.actor.flags &= ~ACTOR_FLAG_20;
+        BgIkanaMirror_SetupWait(this);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Mirror/BgIkanaMirror_Init.s")
+void BgIkanaMirror_Update(Actor* thisx, PlayState* play) {
+    BgIkanaMirror* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Mirror/BgIkanaMirror_Destroy.s")
+    this->actionFunc(this, play);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Mirror/func_80B7FA84.s")
+void BgIkanaMirror_Draw(Actor* thisx, PlayState* play) {
+    s32 pad;
+    BgIkanaMirror* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Mirror/func_80B7FA9C.s")
+    OPEN_DISPS(play->state.gfxCtx);
+    func_8012C28C(play->state.gfxCtx);
+    func_8012C2DC(play->state.gfxCtx);
+    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(POLY_OPA_DISP++, gStoneTowerTempleMirrorDL);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Mirror/func_80B7FB84.s")
+    if (this->lightAbsorptionAlpha > 0) {
+        AnimatedMat_Draw(play, this->lightAbsorptionAnimatedTexture);
+        gDPSetPrimColor(POLY_XLU_DISP++, 0, 0x80, 255, 255, 255, this->lightAbsorptionAlpha);
+        gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(POLY_XLU_DISP++, gStoneTowerTempleMirrorLightEmissionDL);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Mirror/func_80B7FBA4.s")
+    if (this->lightEmissionAlpha > 0) {
+        f32 temp_fv0 = this->lightEmissionAlpha * (1.0f / 255.0f);
```

Comment:
This temp can be something like `alphaFraction`, it's the proportion of the full alpha to use.

Reply:
Fixed in next commit to be pushed.